base/inc/AdePT/BVHNavigator.h

```diff
@@ -211,17 +211,18 @@ class BVHNavigator {
   // Relocate a state that was returned from ComputeStepAndNextVolume: It first
   // removes all volumes from the state that were left, and then recursively
   // locates the pushed point in the containing volume.
-  __host__ __device__ static void RelocateToNextVolume(vecgeom::Vector3D<vecgeom::Precision> const &globalpoint,
+  __host__ __device__ static void RelocateToNextVolume(vecgeom::Vector3D<vecgeom::Precision> &globalpoint,
                                                        vecgeom::Vector3D<vecgeom::Precision> const &globaldir,
                                                        vecgeom::NavStateIndex &state)
   {
     // Push the point inside the next volume.
-    vecgeom::Vector3D<vecgeom::Precision> pushed = globalpoint + 1.E-6 * globaldir;
+    static constexpr double kPush = 1.e-8;
+    globalpoint += kPush * globaldir;
```

Comment:
Changing the globalpoint still does a systematic push, even if only when crossing boundaries. The bias may be negligible in most cases, but isn't it enough to use just the locally pushed point for relocation, and preserve the original globalpoint? I know this may create trouble on the next ComputeStep query since the globalpoint will be on boundary and numerically converted on the wrong side, but we could apply kPush just inside the next ComputeStep and then correct the distance before returning it by adding kPush. This would hide the push ensuring that the start point for querying the geometry is passed the boundary, but in case physics wins will be free of the kPush bias on boundary.

Reply:
So the idea is to keep the tracking point numerically on boundary, but inside both Relocate and ComputeStep work with a locally pushed value to get around the rounding errors.

Reply:
I don't think it would be easy because you'd only want to do it after you moved to a new volume, which the next `ComputeStep` doesn't know about. This gets even trickier for the magnetic field propagator, which needs to query the navigator multiple times. Overall, this PR definitely reduces the bias so I think there is merit, for the rest I'm happy to be proven wrong if you can make it work.